File: csharp/src/Creational/Abstract_Factory/BostonConcreteSoupFactory.cs
//BostonConcreteSoupFactory.java - One of Two concrete factories extending the abstract factory

namespace Creational.Abstract_Factory {


using common;

public class BostonConcreteSoupFactory : AbstractSoupFactory {
    public BostonConcreteSoupFactory() {
        factoryLocation = "Boston";
    }
    public new ClamChowder makeClamChowder() {
        return new BostonClamChowder();
    }
    public new FishChowder makeFishChowder() {
        return new BostonFishChowder();
    }
}

}

File: scala/src/main/scala/Creational/Abstract_Factory/BostonConcreteSoupFactory.scala
//BostonConcreteSoupFactory - One of Two concrete factories extending the abstract factory

package Creational.Abstract_Factory;


import common.*;

public class BostonConcreteSoupFactory extends AbstractSoupFactory {
    public BostonConcreteSoupFactory() {
        factoryLocation = "Boston";
    }
    public ClamChowder makeClamChowder() {
        return new BostonClamChowder();
    }
    public FishChowder makeFishChowder() {
        return new BostonFishChowder();
    }
}
